lib/puppet/pops/types/type_parser.rb

```diff
@@ -140,6 +140,7 @@ def self.type_map
         'iterable'     => TypeFactory.iterable,
         'iterator'     => TypeFactory.iterator,
         'string'       => TypeFactory.string,
+        'sensitive'    => TypeFactory.sensitive,
```

Comment:
In addition to this, since we want to be able to create parameterized Sensitive type instances, for example `Sensitive[Numeric]`, the method `interpret_AccessExpression` must also be modified with an entry for `'sensitive'`

something like this:
```
    when 'sensitive'
      if parameters.size != 1
        raise_invalid_parameters_error('Sensitive', 1, parameters.size)
      end
      param = parameters[0]
      assert_type(ast, param)
      TypeFactory.sensitive(param)
```